tests/parity_client/mod.rs

```diff
@@ -156,15 +155,20 @@ impl ParityClient {
         self.client.eth().balance(address, None).wait().unwrap()
     }
 
-    pub fn send_data(&self, to: Address, data: Option<Bytes>) -> TransactionReceipt {
+    pub fn send_data(
+        &self,
+        to: Address,
+        data: Option<Bytes>,
+        gas_limit: U256,
+    ) -> TransactionReceipt {
         let result_tx = self
             .client
             .personal()
             .send_transaction(
                 TransactionRequest {
                     from: *PARITY_DEV_ACCOUNT,
                     to: Some(to),
-                    gas: None,
```

Comment:
This is one issue, we do not pass a gas limit when sending data to a contract instead of using our own recommended gas limits.